### PR TITLE
Document List | Fixed spacing for filtered message

### PR DESCRIPTION
**Pull request by dannysellers**
Connects #2799

### CODE REVIEW

**Comment by gnakm** [comment]
@dannysellers would you be able to provide a screenshot?

**Comment by dannysellers** [comment]
@gnakm yes! good point
screen shot 2017-10-12 at 12 28 15 pm
screen shot 2017-10-12 at 12 28 37 pm

**Comment by gnakm** [comment]
lgtm!

File: client/app/reader/HeaderFilterMessage.jsx [hunk 1]
```diff
@@ -20,21 +20,22 @@ class HeaderFilterMessage extends React.PureComponent {
...
-    const filteredCategories = _.compact([
+    let filteredCategories = _.compact([
       categoryCount && `Categories (${categoryCount})`,
       tagCount && `Issue tags (${tagCount})`,
       props.viewingDocumentsOrComments === 'comments' && 'Comments'
     ]).join(', ');
 
     if (!filteredCategories.length) {
-      return null;
+      filteredCategories = 'None';
```
Comment (mdbenjam) [inline client/app/reader/HeaderFilterMessage.jsx]: I think this is outside the scope of the ticket. From Gina's mock it looks like there should be no text here. @gnakm can give her thoughts, but we should probably stick with the no-text solution.

File: app/assets/stylesheets/reader/_document_list.scss [hunk 2]
```diff
@@ -42,6 +42,7 @@ $tag-backgrond-color: #d6d7d9;
 
   .documents-table {
     min-width: 810px;
+    margin-top: 0;
```
Comment (mdbenjam) [inline app/assets/stylesheets/reader/_document_list.scss]: I'm having a hard time telling if the spacing is 20px above and below. I'd just check with @gnakm  before committing to make sure the spacing looks good to her. You might need to change the AC if it's not 20, but she signs off on it. Otherwise Artem will measure it and fail it :). He has a hawkeye.

File: client/app/reader/HeaderFilterMessage.jsx (hunk 1 above)
Comment (dannysellers) [reply client/app/reader/HeaderFilterMessage.jsx]: It definitely is outside the scope of fixing spacing between the elements, but it seemed relevant given that the filter text is now displayed permanently. The alternative, to hide the filter element but preserve its vertical spacing, would require refactoring a good amount of CSS. I'd certainly be willing to go down that path, though

File: app/assets/stylesheets/reader/_document_list.scss (hunk 2 above)
Comment (dannysellers) [reply app/assets/stylesheets/reader/_document_list.scss]: Yeah, I tried measuring it using the `addClientRectsOverlay` function in [this MDN article](https://developer.mozilla.org/en-US/docs/Web/API/Element/getClientRects#Example)

**Comment by dannysellers** [comment]
out

File: client/app/reader/HeaderFilterMessage.jsx [hunk 3]
```diff
@@ -20,17 +20,19 @@ class HeaderFilterMessage extends React.PureComponent {
     const categoryCount = getTruthyCount(props.docFilterCriteria.category);
     const tagCount = getTruthyCount(props.docFilterCriteria.tag);
 
-    const filteredCategories = _.compact([
+    let filteredCategories = _.compact([
```
Comment (mdbenjam) [inline client/app/reader/HeaderFilterMessage.jsx]: nit: Since we're not changing this variable now let's switch it back to const.

File: client/app/reader/HeaderFilterMessage.jsx [hunk 4]
```diff
@@ -20,17 +20,19 @@ class HeaderFilterMessage extends React.PureComponent {
...
       categoryCount && `Categories (${categoryCount})`,
       tagCount && `Issue tags (${tagCount})`,
       props.viewingDocumentsOrComments === 'comments' && 'Comments'
     ]).join(', ');
 
+    let cls = 'document-list-filter-message';
```
Comment (mdbenjam) [inline client/app/reader/HeaderFilterMessage.jsx]: nit: Let's use a more descriptive variable name here.
